tsconfig.json

```diff
@@ -0,0 +1,12 @@
+{
+  "compilerOptions": {
+    "outDir": "lib",
+    "allowJs": true,
+    "target": "es6",
```

Comment:
We could use `target: es2017` because thats the best fit to support at least node 8

Reply:
Sure we can change it, I just left any TS optimizations until the end.